ui2/src/pages/QuickSetup/QuickSetup.tsx

```diff
@@ -179,113 +95,193 @@ const QuickSetup = () => {
     deployAllTriggers();
   }, [parties, adminCredentials]);
 
-  return (
-    <div className="quick-setup">
-      <div className="page-controls">
-        <Button className="ghost dark control-button" onClick={() => history.push('/login')}>
-          <ArrowLeftIcon color={'white'} />
-          Back
-        </Button>
-        <Button className="ghost dark control-button" onClick={() => setActiveMenuItem(undefined)}>
-          Skip to Log In
-          <ArrowRightIcon color={'white'} />
-        </Button>
-      </div>
+  let activePage;
 
-      <div className="quick-setup-header">
-        <h1 className="logo-header">
-          <OpenMarketplaceLogo size="32" /> Daml Open Marketplace
-        </h1>
-        {activeMenuItem ? <h2>Market Set-Up</h2> : <h2>Log In</h2>}
-      </div>
-      <div className="quick-setup-tile">
-        {activeMenuItem && (
-          <Menu pointing secondary className="quick-setup-menu page-row">
-            {Object.values(MenuItems).map((item, i) => (
-              <>
-                <Menu.Item
-                  key={i}
-                  disabled={checkIsDisabled(item)}
-                  active={activeMenuItem === item}
-                  onClick={() => setActiveMenuItem(item)}
-                >
-                  <p className={classNames({ visited: !checkIsDisabled(item) })}>{item}</p>
-                </Menu.Item>
-                {i + 1 !== Object.values(MenuItems).length && (
-                  <ArrowRightIcon
-                    color={checkIsDisabled(Object.values(MenuItems)[i + 1]) ? 'grey' : 'blue'}
-                  />
-                )}
-              </>
-            ))}
-          </Menu>
-        )}
-        {activeMenuItem === MenuItems.ADD_PARTIES ? (
-          <AddPartiesPage
-            localOperator={localCreds.party}
-            onComplete={() => setActiveMenuItem(MenuItems.SELECT_ROLES)}
-          />
-        ) : adminCredentials ? (
-          <WellKnownPartiesProvider>
+  if (activeMenuItem === MenuItems.ADD_PARTIES) {
```

Comment:
So there is certainly only ever one active page in the DOM, and makes return code more readable